src/main/scala/encry/view/actors/HistoryApplicator.scala

```diff
@@ -21,135 +22,161 @@ import encry.view.history.History
+      history.reportModifierIsValid(modifier)
+      influxRef.foreach { ref =>
+        ref ! ModifierAppendedToState(success = true)
+        modifier match {
+          case Block(_, payload) if history.isFullChainSynced => ref ! TransactionsInBlock(payload.txs.size)
+          case _ => //do nothing
+        }
+      }
+
+    case NeedToReportAsInValid(block) =>
+      logger.info(s"History got message NeedToReportAsInValid for block ${block.encodedId}.")
+      currentNumberOfAppliedModifiers -= 1
+      val (_, newProgressInfo: ProgressInfo) = history.reportModifierIsInvalid(block)
+      sender() ! NewProgressInfoAfterMarkingAsInValid(newProgressInfo)
+
     case NotificationAboutSuccessfullyAppliedModifier =>
       if (history.isFullChainSynced) {
-        logger.info(s"BlockChain is synced on state applicator at height ${history.getBestHeaderHeight}!")
+        logger.info(s"Block chain is synced at height ${history.getBestHeaderHeight}.")
         ModifiersCache.setChainSynced()
         context.system.eventStream.publish(FullBlockChainIsSynced())
       }
       currentNumberOfAppliedModifiers -= 1
       logger.info(s"Get NotificationAboutSuccessfullyAppliedModifier. Trying to get new one." +
-        s"new currentNumberOfAppliedModifiers is $currentNumberOfAppliedModifiers." +
+        s" New currentNumberOfAppliedModifiers is $currentNumberOfAppliedModifiers." +
         s" ModCache size is ${ModifiersCache.size}.")
       if (modifiersQueue.nonEmpty) {
         logger.info(s"modifiersQueue.nonEmpty in NotificationAboutSuccessfullyAppliedModifier. Sent NewModifierNotification")
         sender() ! NotificationAboutNewModifier
       }
-      getModifierForApplying()
-
-    case NeedToReportAsInValid(block) =>
-      logger.info(s"History got message NeedToReportAsInValid for block ${block.encodedId}.")
-      currentNumberOfAppliedModifiers -= 1
-      val (_, newProgressInfo: ProgressInfo) = history.reportModifierIsInvalid(block)
-      sender() ! NewProgressInfoAfterMarkingAsInValid(newProgressInfo)
+      getModifierForApplying(history)
 
     case nonsense => logger.info(s"History applicator actor got from $sender message $nonsense.")
   }
 
-  def getModifierForApplying(): Unit = if (currentNumberOfAppliedModifiers < setting.levelDB.maxVersions) {
-    logger.debug(s"It's possible to append new modifier to history. Trying to get new one from the cache.")
-    if (locallyGeneratedModifiers.nonEmpty) locallyGeneratedModifiers.dequeueOption.foreach {
-      case (modifier, newQueue) =>
-        locallyGeneratedModifiers = newQueue
+  def getModifierForApplying(history: History): Unit =
+    if (currentNumberOfAppliedModifiers < settings.levelDB.maxVersions) {
+      logger.debug(s"Now it's possible to append new modifier to history. Trying to get new one from the cache.")
+      if (locallyGeneratedModifiers.nonEmpty) locallyGeneratedModifiers.dequeueOption.foreach {
```

Comment:
Why would you first check that `locallyGeneratedModifiers` is non empty, and then dequeue option, if you can omit it?